client/src/pages/Home/Home.jsx

```diff
@@ -0,0 +1,83 @@
+import classNames from 'classnames';
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import './Home.css';
+
+export const Home = ({ username, setUsername, room, setRoom, socket, rooms, setRooms }) => {
+  const navigate = useNavigate();
+
+  const joinRoom = () => {
+    if (room && username) {
+      socket.emit('join_room', { username, room });
+
+      const isRoom = rooms.find(val => val.title === room);
+      const id = [...rooms].sort((a, b) => b.id - a.id)[0].id + 1;
+
+      if (!isRoom) {
+        setRooms(prev => {
+          return [
+            ...prev,
+            {
+              createdAt: Date.now(),
+              id,
+              title: room,
+              updatedAt: Date.now(),
+            }]
+        });
+      }
+
+      navigate('/chat', { replace: true });
+    }
+  };
+
+  const isDisabled = !username || !room;
+
+  return (
+    <div className="container">
+      <div className="formContainer">
+        <h1>{`ChatRooms`}</h1>
+        <input
+          className="input"
+          placeholder='Username...'
+          onChange={(e) => {
+            setUsername(e.target.value)
+            localStorage.setItem('user', e.target.value);
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)